src/main/java/org/sonarlint/intellij/util/ConfigurationSharing.kt

```diff
@@ -0,0 +1,145 @@
+/*
+ * SonarLint for IntelliJ IDEA
+ * Copyright (C) 2015-2024 SonarSource
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.util
+
+import com.intellij.notification.NotificationType
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.MessageDialogBuilder.Companion.okCancel
+import java.io.BufferedWriter
+import java.io.FileWriter
+import java.io.IOException
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import org.jetbrains.annotations.NonNls
+import org.jetbrains.annotations.SystemIndependent
+import org.sonarlint.intellij.common.ui.SonarLintConsole
+import org.sonarlint.intellij.common.util.SonarLintUtils
+import org.sonarlint.intellij.core.BackendService
+import org.sonarlint.intellij.core.BackendService.Companion.projectId
+import org.sonarlint.intellij.core.ProjectBindingManager
+import org.sonarlint.intellij.documentation.SonarLintDocumentation
+import org.sonarlint.intellij.notifications.SonarLintProjectNotifications.Companion.get
+import org.sonarsource.sonarlint.core.rpc.protocol.backend.connection.config.GetSharedConnectedModeConfigFileParams
+import org.sonarsource.sonarlint.core.rpc.protocol.backend.connection.config.GetSharedConnectedModeConfigFileResponse
+
+class ConfigurationSharing {
+
+    companion object {
+        @JvmStatic
+        fun exportConfiguration(project: Project?) {
+            if (project == null) return
+
+            if (confirm(project)) {
+                val configScopeId: @SystemIndependent @NonNls String = projectId(project) ?: return
+                createFile(configScopeId, project)
+            }
+        }
+
+        private fun createFile(configScopeId: String, project: Project) {
+            if (project.getBasePath() == null) {
+                return
+            }
+            SonarLintUtils.getService<BackendService>(BackendService::class.java)
+                .getSharedConnectedModeConfigFileContents(GetSharedConnectedModeConfigFileParams(configScopeId))
```

Comment:
In `getSharedConnectedModeConfigFileContents` you should pass the configscope ID as parameter, and inside the method you should  create the `GetSharedConnectedModeConfigFileParams`